contracts/PodCastNFT.sol

```diff
@@ -5,23 +5,53 @@ import "hardhat/console.sol";
 import "@openzeppelin/contracts/token/ERC721/extensions/ERC721URIStorage.sol";
 import "@openzeppelin/contracts/access/Ownable.sol";
 import "@openzeppelin/contracts/utils/Counters.sol";
-
+import {IERC20} from "./interface/IERC20.sol";
 import {Base64} from "./libraries/Base64.sol";
 
 contract PodCastNFT is ERC721URIStorage, Ownable {
     using Counters for Counters.Counter;
     Counters.Counter private _tokenIds;
-
-    constructor() ERC721("Henkaku v0.2", "henkaku") {}
+    IERC20 public henkakuToken;
+    uint256 public price;
 
     mapping(address => string[]) public roles;
     mapping(uint256 => bool) private _communityMemberShip;
 
+    event BoughtMemberShipNFT(address _owner, uint256 _amount);
+
+    constructor(address _erc20) ERC721("Henkaku v0.2", "henkaku") {
+        henkakuToken = IERC20(_erc20);
+        setPrice(1000e18);
+    }
+
+    function setPrice(uint256 _price) public onlyOwner {
```

Comment:
We might be able to cut it out to a separate ISSUE, but we might want to write a test.

My understanding is that this is to change the price later, but it is unlikely to be tested manually during development.

Reply:
sure! I will write test

Reply:
done